### ISSUE TITLE
Custom learning rate scheduler LambdaLR only supports epoch as argument rather than step

### ISSUE THREAD

**Issue by saahiluppal** [pytorch/pytorch#45895]
I want to implement the following learning rate
```bash
lrate = d_model ^ 0.5 * min( step_num ^ 0.5, step_num * warmup_steps ^ -1.5)
```
And LambdaLR would be most suitable, but how to extract this step_num using LambdaLR?

Thanks

cc @vincentqb

**Comment by saahiluppal** [comment]
Eventually answering my own question
Have a look at this https://huggingface.co/transformers/main_classes/optimizer_schedules.html#schedules
